internal/integration/talos_test.go

```diff
@@ -636,9 +638,11 @@ func AssertTalosUpgradeIsCancelable(testCtx context.Context, st state.State, clu
 
 		events := make(chan state.Event)
 
-		require.NoError(t, st.WatchKind(ctx, omni.NewClusterMachineStatus(resources.DefaultNamespace, "").Metadata(), events),
-			state.WithLabelQuery(resource.LabelEqual(omni.LabelCluster, clusterName)),
-		)
+		t.Logf("watching for the machines in cluster %q", clusterName)
+
+		require.NoError(t, st.WatchKind(ctx, omni.NewClusterMachineStatus(resources.DefaultNamespace, "").Metadata(), events,
+			state.WatchWithLabelQuery(resource.LabelEqual(omni.LabelCluster, clusterName)),
+		))
```

Comment:
oh, I like errors like this. The difference is so subtle: the brackets were placed incorrectly, so `state.WatchWithLabelQuery(resource.LabelEqual(omni.LabelCluster, clusterName)` ended up as the second argument to `require.NoError`, so the watch was actually ignoring the labels 🤦

Reply:
wow :D